src/script/bitcoinconsensus.cpp

```diff
@@ -99,7 +99,13 @@ static int verify_script(const uint8_t *scriptPubKey,
         // Regardless of the verification result, the tx did not error.
         set_error(err, bitcoinconsensus_ERR_OK);
 
-        PrecomputedTransactionData txdata(tx);
+        // Note: The new BIP341 sighash requires knowing all CTxOut of the tx's
```

Comment:
My feedback is basically, if you do the piping first, nobody will be able to make verification code that doesn't work based on this.